### ISSUE TITLE
SonarQube reports coverage on *.cls with test cases but not *.p files.

### ISSUE THREAD

**Issue by kenherring** [Riverside-Software/sonar-openedge#894]
Hey Gilles!  We have test cases in both *.cls file and *.p files.  It appears in sonar that the *.cls files show coverage, but the *.p files do not.  Here's a screenshot for context:

![image](https://user-images.githubusercontent.com/3201462/126002986-9aa2a67e-c0ac-4b13-906c-482acb088f06.png)

I would note that these *.p test programs call other *.p files which DO show coverage.  These test programs are executing and are visible in the profiler output.

I'm not sure if this is a bug or a possible configuration issue.  Do you have any insight on what I could do to confirm one way or the other?

**Comment by gquerret** [comment]
Code coverage requires rcode to be present for each procedure or class, so that's the first thing that you can verify. 
You can also execute the analysis in verbose mode (-v when using Ant Scanner or -X when using command line scanner) ; that will display code coverage information for each file. It can be extremely verbose, so you may want to execute the analysis on a subset of your product first.

**Comment by kenherring-rkt** [comment]
Thanks @gquerret!  I think we figured this out with your help.  When we turned on the `-X` verbose mode we did indeed see that it couldn't find the r-code for coverage.  Here's why:

* When using a fileset to find programs with test cases it returns a fully qualified path.  This example would return `/basedir/somepath/unit/tcTestCase.p` and that file would run via ABLUnit.
  * Here is the fileset directive we use:  `<fileset dir="/basedir" includes="**/unit/tc*" />`
* Our r-code is not in the same directory.  Instead we would find it at `/basedir/obj/somepath/unit/tcTestCase.r`.
* This did not affect classes because it uses the r-code lookup based on the class name rather than the full path name.  Example:  `/basedir/somepath/unit/tcTestClass.cls` would start by searching for a class named "somepath.unit.tcTestClass" in the propath.

I haven't had time to make a simple example yet, but I suspect this is something that can be recreated and fixed pretty easily.  Our workaround was to use `sed` to remove references from the .out files before analysis.
```
sed -i 's,/basedir/,,g' /root/project/artifacts/profiler-*.prof
mvn sonar:sonar
```
